_posts/2023-09-18-confluent-kafka.md

```diff
@@ -0,0 +1,353 @@
+
+Once a connector is deployed and subscribed to one or more topics, then all you need to do is to update a function with an extra annotation. So if you added the `topic: payment.created` annotation to the `new-payment` function, from there on it would be invoked with the payload of every message that was published to the `payment.created` topic.
+
+## Common options for authentication
+
+There are several common ways to configure Apache Kafka, then a few more esoteric options used by some enterprise companies.
+
+In order of complexity:
+
+1. No authentication, plaintext communication (without TLS)
+2. Plaintext communication (without TLS), but with SASL
+3. TLS with a client certificate
+4. TLS without a client certificate, but with SASL
+
+In general, hosted providers will always enable TLS, then use either 3) a client certificate or 4) SASL for authentication.
+
+Confluent Cloud uses TLS and SASL. SASL is a username and password. This is the option we'll be using here, and means creating two secrets in Kubernetes, one for the username, and one for the password.
+
+Aiven uses TLS plus a client certificate using their own self-signed CA, which means creating three secrets, one for the CA, one for the client certificate and one for the private key.
+
+The argument that we tend to hear for 1 or 2, is that a team may be running their stack within a private network or VPC. This model only provides the illusion of security, and is not recommended. It can mean that an eavesdropper or malware that is running within the private environment could potentially gain access to the Kafka cluster.
+
+## Provision a Kafka cluster with Confluent Cloud
+
+Head over to [Confluent Cloud](https://www.confluent.io/en-gb/confluent-cloud/) and sign up as a new customer.
+
+Click on Environments and Default, if Default is not displayed, create it.
+
+Click Clusters or "Add Cluster"
+
+![Pick a region, and understand the plan's limits](/images/2023-09-kafka-confluent/create-cluster.png)
+
+At time of writing, the Basic tier of cluster is free, and more than suitable for testing out the Kafka Connector to see how it works.
+
+If you are concerned about being charged by Confluent for your testing, then pay close attention to any limits or quotas that you may exceed.
+
+Pick a cloud from AWS to Google Cloud to Azure, if you already use one of these vendors, use that one so that you can keep your data in the same region.
+
+![Details about your cluster](/images/2023-09-kafka-confluent/created-cluster.png)
+
+### Obtain connection information
+
+Click *Cluster Overview*, then *API Keys* and *Create Key*
+
+For testing, Confluent recommend using the *Global access* key.
+
+Save the Key as "kafka-broker-username.txt"
+
+Save the Secret as "kafka-broker-password.txt"
+
+Add a description such as "openfaas" or "kafka-connector"
+
+Next, click *Cluster Settings, and under *Endpoints* copy the *Bootstrap server* value.
```

Comment:
Cursive is not rendered correctly because of a missing `*`

```suggestion
Next, click *Cluster Settings*, and under *Endpoints* copy the *Bootstrap server* value.
```